internal/ui/src/components/id.tsx

```diff
@@ -0,0 +1,65 @@
+"use client";
+import { TaskChecked, TaskUnchecked } from "@unkey/icons";
+import * as React from "react";
+import { cn } from "../lib/utils";
+import { Tooltip, TooltipContent, TooltipTrigger } from "./tooltip";
+
+type IdProps = {
+  value: string;
+  truncate?: number;
+  className?: string;
+};
+
+export const Id: React.FC<IdProps> = ({ className, value, truncate, ...props }) => {
+  const [isCopied, setIsCopied] = React.useState(false);
+  const copyTextToClipboard = async (value: string) => {
+    try {
+      await navigator.clipboard.writeText(value ?? "");
+      setIsCopied(true);
+    } catch (error) {
+      console.error("Failed to copy: ", error);
+    }
+  };
+
+  React.useEffect(() => {
+    if (!isCopied) {
+      return;
+    }
+    const timer = setTimeout(() => {
+      setIsCopied(false);
+    }, 2000);
+    return () => clearTimeout(timer);
+  }, [isCopied]);
+
+  const ellipse = "••••";
+  const truncateValue = truncate ? value?.slice(0, truncate) + ellipse : value;
+  const Comp = "button";
```

Comment:
why don't we just use a <button> ?

Reply:
We can do that. Did not think about it for some reason.